error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -794,6 +794,40 @@ Flux<S> after(Flux<T> flux) {
     }
   }
 
+  /** Prefer {@link Mono#ofType(Class)} over more contrived alternatives. */
+  static final class MonoOfType<T, S> {
+    @BeforeTemplate
+    Mono<S> before(Mono<T> mono) {
+      return mono.filter(
+              Refaster.anyOf(
+                  Refaster.<S>clazz()::isInstance,
```

Comment:
Instead of `Refaster.<S>clazz()` we can use a `Class<S> clazz` parameter, in which case non-literal class references are also captured.